apps/niftest/niftest.cpp

```diff
@@ -0,0 +1,90 @@
+///Program to test .nif files both on the FileSystem and in BSA archives.
+
+#include <iostream>
+#include <fstream>
+
+#include <components/nif/niffile.hpp>
+#include <components/files/constrainedfilestream.hpp>
+#include <components/vfs/manager.hpp>
+#include <components/vfs/bsaarchive.hpp>
+
+
+///See if the file has the named extension
+bool hasExtension(std::string filename, std::string  extensionToFind)
+{
+    std::string extension = filename.substr(filename.find_last_of(".")+1);
+
+    //Convert strings to lower case for comparison
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+    std::transform(extensionToFind.begin(), extensionToFind.end(), extensionToFind.begin(), ::tolower);
+
+    if(extension == extensionToFind)
+        return true;
+    else
+        return false;
+}
+
+///See if the file has the "nif" extension.
+bool isNIF(std::string filename)
+{
+    return hasExtension(filename,"nif");
+}
+///See if the file has the "bsa" extension.
+bool isBSA(std::string filename)
+{
+    return hasExtension(filename,"bsa");
+}
+
+///Check all the nif files in the given BSA archive
+void readBSA(std::string filename)
+{
+    VFS::Manager myManager(false);
+    //This does not work, since manager tries to delete it on cleanup
+        //VFS::BsaArchive myBsa(filename);
+        //myManager.addArchive(&myBsa);
```

Comment:
Sorry, habit.
It's a note to self that "take's ownership" means I shouldn't feed the function local variables.
Would you like me to rebase and remove it?